lib/signaling/v2/peerconnection.js

```diff
@@ -940,6 +940,18 @@ class PeerConnectionV2 extends StateMachine {
     });
   }
 
+  /**
+   * we sometimes replace media stream after publishing the track. In such cases published tracks id won't
+   * match the id of the mediaStream track, Fix it here to match original track id.
+   * @private
+   * @param {RTCSessionDescription} description
+   * @return {RTCSessionDescription}
+   */
+  _mapTrackIdToOriginalTrackId(id) {
+    const mediaTrackSender = [...this._rtpSenders.keys()].find(ms => ms.track.id === id);
+    return mediaTrackSender ? mediaTrackSender.id : id;
+  }
```

Comment:
```suggestion
  /**
   * Get the MediaTrackSender ID of the given MediaStreamTrack ID.
   * Since a MediaTrackSender's underlying MediaStreamTrack can be
   * replaced, the corresponding IDs can mismatch.
   * @private
   * @param {Track.ID} id
   * @return {Track.ID}
   */
  _getMediaTrackSenderId(trackId) {
    const mediaTrackSender = Array.from(this._rtpSenders.keys()).find(({ track: { id } }) => id === trackId);
    return mediaTrackSender ? mediaTrackSender.id : trackId;
  }
```